backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresExtension.java

```diff
@@ -114,23 +114,22 @@ private void querySettingExtension() throws IOException, InterruptedException {
         PG_CONTAINER.execInContainer("psql", "-U", selectedDatabase.dbUser(), selectedDatabase.dbName(), "-c", String.format("CREATE EXTENSION IF NOT EXISTS hstore SCHEMA %s;", selectedDatabase.schema()));
     }
 
-    private void initPostgresSession() throws URISyntaxException {
+    private void initPostgresSession() {
         postgresConfiguration = PostgresConfiguration.builder()
-            .url(new URIBuilder()
-                .setScheme("postgresql")
-                .setHost(getHost())
-                .setPort(getMappedPort())
-                .setUserInfo(selectedDatabase.dbUser(), selectedDatabase.dbPassword())
-                .build()
-                .toString())
             .databaseName(selectedDatabase.dbName())
             .databaseSchema(selectedDatabase.schema())
+            .host(getHost())
+            .port(getMappedPort())
+            .username(Optional.of(selectedDatabase.dbUser()))
+            .password(Optional.of(selectedDatabase.dbPassword()))
+            .nonRLSUser(Optional.of(selectedDatabase.dbUser()))
+            .nonRLSPassword(Optional.of(selectedDatabase.dbPassword()))
             .rowLevelSecurityEnabled(rlsEnabled)
             .build();
 
         connectionFactory = new PostgresqlConnectionFactory(PostgresqlConnectionConfiguration.builder()
-            .host(postgresConfiguration.getUri().getHost())
-            .port(postgresConfiguration.getUri().getPort())
+            .host(postgresConfiguration.getHost())
+            .port(postgresConfiguration.getPort())
             .username(postgresConfiguration.getCredential().getUsername())
             .password(postgresConfiguration.getCredential().getPassword())
             .database(postgresConfiguration.getDatabaseName())
```

Comment:
Should we add a new method for PostgresExtension:
`getNonRLSPostgresExecutor`?

tldr:

I see the `PostgresMessageBlobReferenceSourceTest` is tested with `withoutRowLevelSecurity` 
I guess it will not pass if enable `withRowLevelSecurity` + insert into the messages table with some random domain. Then `blobReferenceSource` will not retrieve full lists

Reply:
We should do this in integration tests